airbyte-cdk/python/airbyte_cdk/sources/file_based/config/file_based_stream_config.py

```diff
@@ -101,10 +90,22 @@ def _transform_legacy_parquet_or_avro_config(cls, config: Mapping[str, Any]) ->
         return {**config, **{"decimal_as_float": True}}
 
     @validator("input_schema", pre=True)
-    def transform_input_schema(cls, v: Optional[Union[str, Mapping[str, Any]]]) -> Optional[Mapping[str, Any]]:
+    def validate_input_schema(cls, v: Optional[str]) -> Optional[str]:
```

Comment:
as per the PR description, it feels like a bit of a code smell that we hijack Pydantic validators to do some under the hood transformations. It forces us to loosen the interface when we really only want the user to specify `input_schema` as a string. By just doing validation here, we can make the Pydantic model represent only the format the user should be using, not our underlying implementation